posts/2021-06-11-request-timing-21006.adoc

```diff
@@ -0,0 +1,193 @@
+----
+<dependency>
+    <groupId>io.openliberty</groupId>
+    <artifactId>openliberty-runtime</artifactId>
+    <version>21.0.0.6</version>
+    <type>zip</type>
+</dependency>
+----
+
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[21.0.0.6,)'
+}
+----
+
+Or if you're using Docker:
+
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+
+// // // // // // // //
+// Above:
+// Replace the RELEASE_VERSION with the Open Liberty release number e.g. 20.0.0.4
+// // // // // // // //
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+
+[#metrics]
+=== Request Timing now supported by MicroProfile Metrics
+
+The request timing feature (`requestTiming-1.0`) is used to keep track of the slow and hung requests for servlet requests with a RequestTimingStats MXBean. The Microprofile Metrics features (`mpMetrics-X.X`) on the other hand provides vendor metrics that are distinct to the Open Liberty runtime.
+
+Starting with the Open Liberty 21.0.0.6 the RequestTimingStats MXBean will now have it’s data retrieved by the MicroProfile Metrics feature for reporting on the `/metrics` (or `/metrics/vendor`) endpoint. This functionality is compatible with both the 2.X and 3.X MicroProfile Metrics features.
+
+Below is a sample output of the new request timing metrics:
+
+[source, xml]
+----
+# TYPE vendor_requestTiming_activeRequestCount gauge
+# HELP vendor_requestTiming_activeRequestCount The number of servlet requests currently running.
+vendor_requestTiming_activeRequestCount 1
+
+# TYPE vendor_requestTiming_requestCount_total counter
+# HELP vendor_requestTiming_requestCount_total The number of servlet requests since the server started.
+vendor_requestTiming_requestCount_total 3
+
+# TYPE vendor_requestTiming_hungRequestCount gauge
+# HELP vendor_requestTiming_hungRequestCount The number of servlet requests that are currently running but are hung.
+vendor_requestTiming_hungRequestCount 0
+
+# TYPE vendor_requestTiming_slowRequestCount gauge
+# HELP vendor_requestTiming_slowRequestCount The number of servlet requests that are currently running but are slow.
+vendor_requestTiming_slowRequestCount 0
+----
+
+To be able to retrieve request timing metrics from `/metrics` you will need to enable both the `requestTiming-1.0` feature in combination with one of the following MicroProfile Metric features: `mpMetrics-2.0`, `mpMetrics-2.2` or `mpMetrics-2.3` or `mpMetric-3.0`.
+
+To begin recieving metrics you will need to configure the request timing thresholds for slow or hung requests. The following example demonstrates enabling request timing metrics for `mpMetrics-3.0`, with a sample threshold configuration for both slow and hung servlet requests.
```

Comment:
Change this sentence to:  "The metrics you receive depend on the configuration of the request timing thresholds for slow or hung requests."

Reply:
Done